### ISSUE TITLE
Users that are not logged-in to HumHub are still able to access documents

### ISSUE THREAD

**Issue by WizWareTech** [ONLYOFFICE/onlyoffice-humhub#84]
I can see how this could be convenient in some cases, but it is a huge security risk for private docs that administrators don't want public users to be able to access.

How can administrators limit document sharing to only users that are logged in (and possibly only to members of the Space where the file is stored?)

Ideally, there should be another permission or permissions added to spaces that controls this behavior for shared files.

In the meantime, how can I go about disabling file sharing via links all-together so users don't expose private files inside a Space to the public? (I don't mind editing some PHP if needed, I just need to know where/what to edit). And thank you...the product is working well so far as we test it for production!

**Comment by nicochto** [comment]
@WizWareTech I totally agree with you, this sharing feature is very problematic. Here is a dirty workaround to remove the button: Open the file /var/www/humhub/protected/modules/onlyoffice/widgets/views/editor.php and at line 30 you replace the entire line:
`            <?= humhub\libs\Html::a(Yii::t('OnlyofficeModule.base', 'Share'), '#', ['class' => 'btn btn btn-default', 'data-action-click' => 'share', 'data-action-block' => 'sync', 'data-action-url' => Url::to(['/onlyoffice/share', 'guid' => $file->guid, 'mode' => $mode])]); ?>`
with this one:
`            <??>`
and there is no more Share button in the editor :-)

But you need to do this again after each update of the Onlyoffice module.
